sympy/polys/polytools.py

```diff
@@ -7938,3 +7937,339 @@ def named_poly(n, f, K, name, x, polys):
+
+    minimal_combinations = []
+    for combo in all_combinations:
+        if not any(combo > other for other in all_combinations):
+            minimal_combinations.append(combo)
+
+    return [[(factor, factor_dict[factor]) for factor in combo]
+            for combo in minimal_combinations]
+
+
+def factor_system_cond(eqs, gens=None, **kwargs):
+    """Factor a system of equations and return factors with symbolic conditions.
+
+    Converts equations to polynomials, factors them, and returns factors with
+    Boolean conditions expressing when each factor is valid. Handles both
+    polynomial and transcendental equations.
+
+    Parameters
+    ==========
+
+    eqs : list
+        List of equations or expressions to be factored. Can include:
+        - Polynomial equations
+        - Transcendental equations
+        - Zero expressions
+        - Constants
+    gens : list, optional
+        List of generators (variables). If None, determined automatically
+        from the equations' free symbols
+    **kwargs : dict
+        Additional keyword arguments for factorization routines
+
+    Returns
+    =======
+
+    list of list of tuple
+        Each inner list represents a solution set.
+        Each tuple contains (factor, condition) where:
+        - factor is a symbolic expression (not a Poly)
+        - condition is either:
+            * sympy.true if no conditions apply
+            * Or(Ne(c1, 0), Ne(c2, 0), ...) for non-zero conditions
+
+    Examples
+    ========
+
+    >>> from sympy import symbols
+    >>> from sympy.polys.polytools import factor_system_cond
+    >>> x, a, b = symbols('x a b')
+    >>> factor_system_cond([(a - 1)*(x - 2), (b - 3)*(x - 2)], [x])
+    [[(x - 2, Ne(a - 1, 0) | Ne(b - 3, 0))]]
+
+    Notes
+    =====
+
+    - Returns [] if input list is empty
+    - Returns [[]] if all expressions are zero, indicating infinite solutions
+    - Returns [] if any non-zero constant is present, indicating no solution exists
+    - Handles transcendental equations that can be converted to polynomial form
+    - Maintains symbolic conditions for factor validity
+    """
+
+    if not eqs:
+        return []
+
+    exprs = [sympify(eq) for eq in eqs]
+    exprs = [expr.simplify() for expr in exprs]
+
+    if all(expr.is_zero for expr in exprs):
+        return [[]]
+
+    if any(expr.is_constant() and not expr.is_zero for expr in exprs):
+        return []
+
+    exprs = [expr for expr in exprs if not expr.is_zero]
+
+    if not exprs:
+        return [[]]
+
+    try:
+        polys, _ = parallel_poly_from_expr(exprs, gens, **kwargs)
+    except GeneratorsNeeded:
+        d = Dummy()
+        try:
+            polys, _ = parallel_poly_from_expr(exprs, [d], **kwargs)
+            if all(poly.is_zero for poly in polys):
+                return [[]]
+            return []
+        except PolificationFailed:
+            return []
+    except PolificationFailed:
+        return []
```

Comment:
Why does this catch `PolificationFailed`? Presumably an error should be raised in this case.